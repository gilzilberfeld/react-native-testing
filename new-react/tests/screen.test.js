import TabOneScreen from '..//screens/TabOneScreen'
import {fireEvent, render} from '@testing-library/react-native'
import NotFoundScreen from '../screens/NotFoundScreen';

describe('screen tests', ()=>{

    it('screen renders to json', ()=>{
        const { toJSON } = render(<TabOneScreen/>); 
        expect(toJSON()).toMatchSnapshot();
    })

    it('screen renders correctly', ()=> {
        const { getByText } = render(<TabOneScreen/>); 
        const tab = getByText('Tab One');
        
        const title = tab.children[0].children[0];
        expect(title).toEqual('Tab One');
    })

    it('search by TestID in component', ()=> {
        const { getByTestId } = render(<TabOneScreen/>); 
        const tab = getByTestId('theTestID');
        const textComponent = tab.props.children;
        expect(textComponent).toEqual('UpperText');
    })

    it('inject mock navigation', ()=>{
        testNavigation = {
            replace : jest.fn()
        }
        const mockReplace = testNavigation.replace;
        
        const { getByTestId } = render(<NotFoundScreen navigation={testNavigation} />); 
        const touchOp = getByTestId('touchable');
        fireEvent(touchOp, 'onPress');

        // Check that replace was called
        expect(mockReplace.mock.calls.length).toBe(1);
        expect(mockReplace).toHaveBeenCalledTimes(1);

        // Check that replace was called with the right arg
        expect(mockReplace.mock.calls[0][0]).toBe('Root');
        expect(mockReplace).toHaveBeenCalledWith('Root');
    })
})