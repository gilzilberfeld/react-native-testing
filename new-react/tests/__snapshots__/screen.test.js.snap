// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen tests screen renders to json 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
      },
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#000",
        },
        Object {
          "fontSize": 20,
          "fontWeight": "bold",
        },
      ]
    }
  >
    Tab One
  </Text>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#eee",
        },
        Object {
          "height": 1,
          "marginVertical": 30,
          "width": "80%",
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
          },
          Object {
            "alignItems": "center",
            "marginHorizontal": 50,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "rgba(0,0,0,0.8)",
            },
            Object {
              "fontSize": 17,
              "lineHeight": 24,
              "textAlign": "center",
            },
          ]
        }
        testID="theTestID"
      >
        UpperText
      </Text>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "rgba(0,0,0,0.05)",
            },
            Array [
              Object {
                "borderRadius": 3,
                "paddingHorizontal": 4,
              },
              Object {
                "marginVertical": 7,
              },
            ],
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000",
              },
              Array [
                undefined,
                Object {
                  "fontFamily": "Courier New",
                  "fontWeight": "500",
                },
              ],
            ]
          }
          testID="theMonoTestID"
        >
          StaticText
        </Text>
      </View>
      <Text
        accessibilityRole="text"
        style={
          Array [
            Object {
              "color": "rgba(0,0,0,0.8)",
            },
            Object {
              "fontSize": 17,
              "lineHeight": 24,
              "textAlign": "center",
            },
          ]
        }
      >
        InnerText
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
          },
          Object {
            "alignItems": "center",
            "marginHorizontal": 20,
            "marginTop": 15,
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "paddingVertical": 15,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#2f95dc",
              },
              Object {
                "textAlign": "center",
              },
            ]
          }
        >
          Tap here if your app doesn't automatically update after making changes
        </Text>
      </View>
    </View>
  </View>
</View>
`;
